### PR TITLE
ticket-1413 SNB thread tries force download connected S-Chains until success when invoked first time at startu

### LINKED ISSUE #1413: Processing chain data work unstable
**Versions**
Skaled: 3.16.0-beta.3(4)
IMA: 1.3.5-beta.0
IMA contract mainnet: ima.1.3.4-rc.0( for both version 1.3.5-beta.0)
Skale Manager : 1.9.2-stable.0

**Environment**
20 active nodes on the network
4 schains up and running 

**Preconditions**
Update process to release versions 

**Steps to Reproduce/Description**
Observe logs after update 

**Expected behavior**
During init of IMA agent SNB should collect all information about other schain each hour: 
2023-01-23 02:39:41.749: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 03:39:43.613: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 04:39:45.392: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 05:39:47.213: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server

**Actual result** 
In some cases SNB init failed, see more logs below. 
2023-01-23 05:39:47.213: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 06:39:49.155: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 08:39:52.559: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
2023-01-23 09:39:54.263: SNB WORKER Full init compete for in-worker SNB server skale_observer_worker_server
Init failed at 07:39 
NOTE: Not related to connect ECONNREFUSED

[SNBFail.txt](https://github.com/skalenetwork/IMA/files/10482018/SNBFail.txt)
[SNBSuccess.txt](https://github.com/skalenetwork/IMA/files/10482019/SNBSuccess.txt)

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1595
```

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
- SNB thread tries force download connected S-Chains until success when invoked first time at startup
- added verified and tested support for both node js 18 and 20, where 18 is current LTS version and 20 will become main LTS version in January 2024
- switched node js version to 18
- fixed "to-boolean" type conversion everywhere it was possible, i.e. converted `something ? true : false` into `!!something` as D1 and D3 suggested

### CODE REVIEW

**Review by kladkogex — DISMISSED** [review]

File: npms/skale-observer/observerWorker.mjs [hunk 1]
```diff
@@ -273,7 +294,12 @@ class ObserverServer extends SocketServer {
                         "\n" );
                 }
                 await self.periodicCachingDoNow(
-                    socket, secondsToReDiscoverSkaleNetwork, strChainNameConnectedTo );
+                    socket,
+                    secondsToReDiscoverSkaleNetwork,
+                    strChainNameConnectedTo,
+                    isForceMultiAttemptsUntilSuccess ? true : false
```
Comment (DmytroNazarenko) [inline npms/skale-observer/observerWorker.mjs]: I think we can remove `? true : false`

File: agent/discoveryTools.mjs [hunk 2]
```diff
@@ -61,7 +61,8 @@ export function initialSkaleNetworkScanForS2S() {
                 "secondsToWaitForSkaleNetworkDiscovered":
                     imaState.optsS2S.secondsToWaitForSkaleNetworkDiscovered,
                 "chain": imaState.chainProperties.sc,
-                "bParallelModeRefreshSNB": imaState.optsS2S.bParallelModeRefreshSNB ? true : false
+                "bParallelModeRefreshSNB": imaState.optsS2S.bParallelModeRefreshSNB ? true : false,
```
Comment (dmytrotkk) [inline agent/discoveryTools.mjs]: could be replaced with `!!imaState.optsS2S.bParallelModeRefreshSNB`

**Review by dmytrotkk — DISMISSED** [review]

File: agent/discoveryTools.mjs (hunk 2 above)
Comment (sergiy-skalelabs) [reply agent/discoveryTools.mjs]: yes, agree

Comment (sergiy-skalelabs) [reply agent/discoveryTools.mjs]: Fixed in many places, thanx

File: npms/skale-observer/observerWorker.mjs (hunk 1 above)
Comment (sergiy-skalelabs) [reply npms/skale-observer/observerWorker.mjs]: Done, replaced everywhere it was possible with !! as D1 suggested

**Review by dmytrotkk — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**